pandas/core/indexes/base.py

```diff
@@ -4544,12 +4544,22 @@ def join(
 
         if len(other) == 0 and how in ("left", "outer"):
```

Comment:
can you make a `if len(other)` and if `len(self)` clause then sub if's here for the cases

Reply:
Done.